### PR TITLE
[INLONG-9831][Agent] Increase configuration acquisition capability

### LINKED ISSUE #9831: [Improve][Agent] Increase configuration acquisition capability
### Description

Increase configuration acquisition capability

### InLong Component

InLong Agent

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by justinwwhuang**
[INLONG-9831][Agent] Increase configuration acquisition capability
- Fixes #9831 

### Motivation

Increase configuration acquisition capability
### Modifications
Increase configuration acquisition capability
### Verifying this change

*(Please pick either of the following options)*

- [ ] This change is a trivial rework/code cleanup without any test coverage.

- [ ] This change is already covered by existing tests, such as:
  *(please describe tests)*

- [ ] This change added tests and can be verified as follows:

  *(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a follow-up issue for adding the documentation

### CODE REVIEW

File: inlong-agent/agent-installer/src/main/java/org/apache/inlong/agent/installer/ManagerResultFormatter.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    /**
+     * get json result
+     *
+     * @return json object
+     */
+    public static JsonObject getResultData(String jsonStr) {
+        JsonObject object = GSON.fromJson(jsonStr, JsonObject.class);
+        if (object == null || !object.has(RESULT_CODE) || !object.has(RESULT_DATA)
+                || !SUCCESS_CODE.equals(object.get(RESULT_CODE).getAsString())) {
+            LOGGER.error("cannot get result data, please check manager status, return str is {}", jsonStr);
```
Comment (aloyszhang) [inline inlong-agent/agent-installer/src/main/java/org/apache/inlong/agent/installer/ManagerResultFormatter.java:45]: All these conditions' purpose is to print the error log, besides the log, they have no effect.
So, IMO, the `warn` level for the log is enough, WDYT?

File: inlong-agent/agent-installer/src/main/java/org/apache/inlong/agent/installer/ManagerFetcher.java [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+            throw new RuntimeException("init manager error, cannot find required key");
+        }
+    }
+
+    private boolean requiredKeys(InstallerConfiguration conf) {
+        return conf.hasKey(MANAGER_ADDR);
+    }
+
+    /**
+     * build config url for manager according to config
```
Comment (aloyszhang) [inline inlong-agent/agent-installer/src/main/java/org/apache/inlong/agent/installer/ManagerFetcher.java:104]: Plz capitalize the first letter, and so do all remaining methods.

File: inlong-agent/agent-installer/src/main/java/org/apache/inlong/agent/installer/ModuleManager.java [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+            return;
+        }
+
+    }
+
+    @Override
+    public void start() throws Exception {
+        LOGGER.info("starting installer manager");
+        submitWorker(coreThread());
+        LOGGER.info("starting agent manager end");
```
Comment (aloyszhang) [inline inlong-agent/agent-installer/src/main/java/org/apache/inlong/agent/installer/ModuleManager.java:101]:
```suggestion
        LOGGER.info("started agent manager");
```

File: inlong-agent/agent-installer/src/main/java/org/apache/inlong/agent/installer/ModuleManager.java [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+    public static final int CONFIG_QUEUE_CAPACITY = 1;
+    public static final int CORE_THREAD_SLEEP_TIME = 1000;
+    private static final Logger LOGGER = LoggerFactory.getLogger(ModuleManager.class);
+    private final InstallerConfiguration conf;
+    private final BlockingQueue<ConfigResult> configQueue;
+
+    private String curMd5;
+
+    public ModuleManager() {
+        conf = InstallerConfiguration.getInstallerConf();
```
Comment (aloyszhang) [inline inlong-agent/agent-installer/src/main/java/org/apache/inlong/agent/installer/ModuleManager.java:48]: conf is never used

File: inlong-agent/agent-installer/src/main/java/org/apache/inlong/agent/installer/ModuleManager.java [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+        configQueue = new LinkedBlockingQueue<>(CONFIG_QUEUE_CAPACITY);
+    }
+
+    public void submitConfig(ConfigResult config) {
+        if (config == null) {
+            return;
+        }
+        while (configQueue.size() != 0) {
+            configQueue.poll();
+        }
```
Comment (aloyszhang) [inline inlong-agent/agent-installer/src/main/java/org/apache/inlong/agent/installer/ModuleManager.java:58]:
```suggestion
         configQueue.clear();
```

File: inlong-agent/agent-installer/src/main/java/org/apache/inlong/agent/installer/ModuleManager.java [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+                            AgentUtils.getCurrentTime(), 1, 1);
+                } catch (Throwable ex) {
+                    LOGGER.error("exception caught", ex);
+                    ThreadUtils.threadThrowableHandler(Thread.currentThread(), ex);
+                }
+            }
+        };
+    }
+
+    private void dealWithConfigQueue(BlockingQueue<ConfigResult> queue) {
```
Comment (aloyszhang) [inline inlong-agent/agent-installer/src/main/java/org/apache/inlong/agent/installer/ModuleManager.java:88]: Is there missing some processing logic for the ConfigResult polled from the queue?

File: inlong-agent/agent-installer/src/main/java/org/apache/inlong/agent/installer/ModuleManager.java [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+ */
+public class ModuleManager extends AbstractDaemon {
+
+    public static final int CONFIG_QUEUE_CAPACITY = 1;
+    public static final int CORE_THREAD_SLEEP_TIME = 1000;
+    private static final Logger LOGGER = LoggerFactory.getLogger(ModuleManager.class);
+    private final InstallerConfiguration conf;
+    private final BlockingQueue<ConfigResult> configQueue;
+
+    private String curMd5;
```
Comment (aloyszhang) [inline inlong-agent/agent-installer/src/main/java/org/apache/inlong/agent/installer/ModuleManager.java:45]: remove this line

File: inlong-agent/agent-installer/src/main/java/org/apache/inlong/agent/installer/ManagerFetcher.java [hunk 8]
```diff
@@ -0,0 +1,223 @@
...
+/**
+ * Fetch command from Inlong-Manager
+ */
+public class ManagerFetcher extends AbstractDaemon implements ProfileFetcher {
+
+    public static final String MANAGER_ADDR = "manager.addr";
+    public static final String MANAGER_AUTH_SECRET_ID = "manager.auth.secretId";
+    public static final String MANAGER_AUTH_SECRET_KEY = "manager.auth.secretKey";
+    public static final String CLUSTER_NAME = "cluster.name";
+    public static final String CLUSTER_TAG = "cluster.tag";
```
Comment (aloyszhang) [inline inlong-agent/agent-installer/src/main/java/org/apache/inlong/agent/installer/ManagerFetcher.java:66]: remove this line

File: inlong-agent/agent-installer/src/main/java/org/apache/inlong/agent/installer/ManagerFetcher.java [hunk 9]
```diff
@@ -0,0 +1,223 @@
...
+     * example - http://127.0.0.1:8080/inlong/manager/openapi/installer/getConfig
+     */
+    private String buildStaticConfigUrl(String baseUrl) {
+        return baseUrl + conf.get(INSTALLER_MANAGER_CONFIG_HTTP_PATH, DEFAULT_INSTALLER_MANAGER_CONFIG_HTTP_PATH);
+    }
+
+    /**
+     * request manager to get commands, make sure it is not throwing exceptions
+     */
+    public ConfigResult getConfig() {
```
Comment (aloyszhang) [inline inlong-agent/agent-installer/src/main/java/org/apache/inlong/agent/installer/ManagerFetcher.java:115]: Is this method needed?

File: inlong-agent/agent-installer/src/main/java/org/apache/inlong/agent/installer/ManagerFetcher.java [hunk 10]
```diff
@@ -0,0 +1,223 @@
...
+     *
+     * @return runnable profile.
+     */
+    private Runnable configFetchThread() {
+        return () -> {
+            Thread.currentThread().setName("ManagerFetcher");
+            while (isRunnable()) {
+                try {
+                    int configSleepTime = conf.getInt(AGENT_FETCHER_INTERVAL, DEFAULT_AGENT_FETCHER_INTERVAL);
+                    ConfigResult config = getTestConfig();
```
Comment (aloyszhang) [inline inlong-agent/agent-installer/src/main/java/org/apache/inlong/agent/installer/ManagerFetcher.java:155]:
```suggestion
            int configSleepTime = conf.getInt(AGENT_FETCHER_INTERVAL, DEFAULT_AGENT_FETCHER_INTERVAL);
                    ConfigResult config = getTestConfig();
            while (isRunnable()) {
                try {
                    
```

**Comment by aloyszhang** [comment]
Left some comments, PTAL.

File: inlong-agent/agent-installer/src/main/java/org/apache/inlong/agent/installer/ModuleManager.java (hunk 4 above)
Comment (justinwwhuang) [reply inlong-agent/agent-installer/src/main/java/org/apache/inlong/agent/installer/ModuleManager.java:48]: This code is not complete yet, we will use it later

File: inlong-agent/agent-installer/src/main/java/org/apache/inlong/agent/installer/ModuleManager.java (hunk 6 above)
Comment (justinwwhuang) [reply inlong-agent/agent-installer/src/main/java/org/apache/inlong/agent/installer/ModuleManager.java:86]: This code is not complete yet, we will use it later

File: inlong-agent/agent-installer/src/main/java/org/apache/inlong/agent/installer/ManagerFetcher.java (hunk 9 above)
Comment (justinwwhuang) [reply inlong-agent/agent-installer/src/main/java/org/apache/inlong/agent/installer/ManagerFetcher.java:115]: This code is not complete yet

File: inlong-agent/agent-installer/src/main/java/org/apache/inlong/agent/installer/ManagerFetcher.java (hunk 10 above)
Comment (justinwwhuang) [reply inlong-agent/agent-installer/src/main/java/org/apache/inlong/agent/installer/ManagerFetcher.java:155]: This configuration needs to be dynamically updated

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/fetcher/ManagerFetcher.java [hunk 11]
```diff
@@ -141,12 +141,12 @@ public TaskResult fetchTaskConfig() {
     public TaskResult getStaticConfig() {
         LOGGER.info("getStaticConfig start");
         String resultStr = httpManager.doSentPost(staticConfigUrl, getFetchRequest(null));
-        LOGGER.info("test123 staticConfigUrl {}", staticConfigUrl);
+        LOGGER.info("staticConfigUrl {}", staticConfigUrl);
```
Comment (dockerzhang) [inline inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/fetcher/ManagerFetcher.java:144]: Add a more readable description.

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/fetcher/ManagerFetcher.java [hunk 12]
```diff
@@ -141,12 +141,12 @@ public TaskResult fetchTaskConfig() {
...
         JsonObject resultData = getResultData(resultStr);
         JsonElement element = resultData.get(AGENT_MANAGER_RETURN_PARAM_DATA);
         LOGGER.info("getStaticConfig end");
         if (element != null) {
-            LOGGER.info("test123 getStaticConfig not null {}", resultData);
+            LOGGER.info("getStaticConfig not null {}", resultData);
```
Comment (dockerzhang) [inline inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/fetcher/ManagerFetcher.java:149]: ditto

File: inlong-agent/agent-installer/src/main/java/org/apache/inlong/agent/installer/ModuleManager.java [hunk 13]
```diff
@@ -0,0 +1,131 @@
...
+        }
+        if (updateModules(config.getModuleList())) {
+            curMd5 = config.getMd5();
+        } else {
+            LOGGER.error("Update modules failed!");
+        }
+    }
+
+    private boolean updateModules(List<ModuleConfig> modules) {
+        return true;
```
Comment (EMsnap) [inline inlong-agent/agent-installer/src/main/java/org/apache/inlong/agent/installer/ModuleManager.java:106]: always return true, what does this method for if it always returns true

File: inlong-agent/agent-installer/src/main/java/org/apache/inlong/agent/installer/ModuleManager.java (hunk 7 above)
Comment (justinwwhuang) [reply inlong-agent/agent-installer/src/main/java/org/apache/inlong/agent/installer/ModuleManager.java:47]: will use it next issue

File: inlong-agent/agent-installer/src/main/java/org/apache/inlong/agent/installer/ManagerFetcher.java (hunk 8 above)
Comment (justinwwhuang) [reply inlong-agent/agent-installer/src/main/java/org/apache/inlong/agent/installer/ManagerFetcher.java:66]: Will use it next issue

File: inlong-agent/agent-installer/src/main/java/org/apache/inlong/agent/installer/ModuleManager.java (hunk 13 above)
Comment (justinwwhuang) [reply inlong-agent/agent-installer/src/main/java/org/apache/inlong/agent/installer/ModuleManager.java:106]: Will use it next issue

**Review by dockerzhang — APPROVED** [review]

**Review by aloyszhang — APPROVED** [review]

**Pull request merged**